File: selectionSort.cpp
#include <iostream>
using namespace std;

void swap(int* a,int* b);
void selectionSort(int arr[],int size);
void printArray(int arr[],int size);
int main()
{
    int arr[]={5,3,7,2,8};
    int size=sizeof(arr)/sizeof(arr[0]);
    selectionSort(arr,size);
    printf("Sorted array : ");
    printArray(arr,size);
    return 0;
}

void printArray(int arr[],int size){
    for(int i=0;i<size;i++){
        printf("%d ",arr[i]);
    }
    printf("\n");
}
void selectionSort(int arr[],int size){
    int min_index;
    for(int i=0;i<size;i++){
        min_index=i;
        for(int j=i+1;j<size;j++){
            if(arr[j]<arr[min_index]){
                min_index=j;
            }
        }
        printArray(arr,size);
        swap(&arr[min_index],&arr[i]);
    }
}

void swap(int* a,int* b){
    int temp=*a;
    *a=*b;
    *b=temp;
}

File: SelectionSort.java
class SelectionSort{
	public static void main(String args[]){
		int[] arr= {4,5,2,3,1};
		selectionSort(arr);
		for(int i=0;i<arr.length;i++){
			System.out.println(arr[i]);
		}
		
	}	
	public static void selectionSort(int[] a){
		for(int i=0;i<a.length;i++){
			int min=i,temp;
			printArr(a);
			for(int j=i;j<a.length;j++){
				if(a[min]>a[j]){
				min=j;
				}
			}

			temp=a[i];
			a[i]=a[min];
			a[min]=temp;
		}
	}

	public static void printArr(int[] a){
		for(int i=0;i<a.length;i++){
			System.out.print(a[i]+" ");
		}
		System.out.println(" ");
	}
}










